eng/docgeneration/docindex.yml

```diff
@@ -0,0 +1,119 @@
+      - pwsh: ls
+        workingDirectory: $(Build.SourcesDirectory)/docfx_project/templates
+      - pwsh: |
+          Copy-Item "$(Build.SourcesDirectory)/eng/docgeneration/docfx.json" -Destination "$(Build.SourcesDirectory)/docfx_project/" -Force
+        displayName: Copy over docfx.json
+      - pwsh: |
+          $ServiceList = Get-ChildItem "$(Build.SourcesDirectory)/sdk" -Directory -Exclude eng, mgmtcommon, template | Sort-Object
+          $YmlPath = "$(Build.SourcesDirectory)/docfx_project/api"
+          New-Item -Path $YmlPath -Name "toc.yml" -Force
+          foreach ($Dir in $ServiceList)
+          {
+              New-Item -Path $YmlPath -Name "$($Dir.Name).md" -Force
+              Write-Host "Write to Toc.yml"
+              Add-Content -Path "$($YmlPath)/toc.yml" -Value "- name: $($Dir.Name.ToUpper())`r`n  href: $($Dir.Name).md"
+              Write-Host "Write Client Header"
+              $PkgList = Get-ChildItem $Dir.FullName -Directory -Exclude .vs, .vscode, Azure.Security.KeyVault.Shared | Where-Object -FilterScript {$_.Name -notmatch ".Management."}
+              if (($PkgList | Measure-Object).count -eq 0)
+              {
+                  continue
+              }
+              Add-Content -Path "$($YmlPath)/$($Dir.Name).md" -Value "# CLIENT"
+              Add-Content -Path "$($YmlPath)/$($Dir.Name).md" -Value "---"
+              foreach ($Pkg in $PkgList)
+              {
+                  $ProjectName = Get-ChildItem "$($pkg.FullName)\src\*" -Include *.csproj
+                  Write-Host "Write $($ProjectName.Name)"
+                  Add-Content -Path "$($YmlPath)/$($Dir.Name).md" -Value "#### $($ProjectName.BaseName)"
+              }
+          }
+          foreach ($Dir in $ServiceList)
+          {
+              Write-Host "Write Mgmt Header"
+              $PkgList = Get-ChildItem $Dir.FullName -Directory | Where-Object -FilterScript {$_.Name -match ".Management."}
+              if (($PkgList | Measure-Object).count -eq 0)
+              {
+                  continue
+              }
+              Add-Content -Path "$($YmlPath)/$($Dir.Name).md" -Value "# MANAGEMENT"
+              Add-Content -Path "$($YmlPath)/$($Dir.Name).md" -Value "---"
+              foreach ($Pkg in $PkgList)
+              {
+                  $ProjectName = Get-ChildItem "$($pkg.FullName)\src\*" -Include *.csproj
+                  Write-Host "Write $($ProjectName.Name)"
+                  Add-Content -Path "$($YmlPath)/$($Dir.Name).md" -Value "#### $($ProjectName.BaseName)"
+              }
+          }
+          New-Item -Path "$(Build.SourcesDirectory)/docfx_project" -Name "toc.yml" -Force
+          Add-Content -Path "$(Build.SourcesDirectory)/docfx_project/toc.yml" -Value "- name: Azure SDK for NET APIs`r`n  href: api/`r`n  homepage: api/index.md"
+          Copy-Item "$(Build.SourcesDirectory)/README.md" -Destination "$(Build.SourcesDirectory)/docfx_project/api/index.md" -Force
```

Comment:
How does the toc and the index get merged together? Is that something that docfx knows about or is there a step somewhere I'm overlooking?

Reply:
The toc.yml gets converted to toc.html by docfx. Then the index.html pulls it in and it becomes navigation.